src/main/groovy/com/adarshr/gradle/testlogger/TestLoggerPlugin.groovy

```diff
@@ -28,20 +30,15 @@ class TestLoggerPlugin implements Plugin<Project> {
     }
 
     private static void createExtensions(Project project) {
-        project.extensions.create('testlogger', TestLoggerExtension, project)
-        project.tasks.withType(Test).each { task ->
-            task.extensions.create('testlogger', TestLoggerExtension, project)
-        }
-        project.tasks.whenTaskAdded { task ->
-            if (task instanceof Test) {
-                task.extensions.create('testlogger', TestLoggerExtension, project)
-            }
+        project.extensions.create(EXTENSION_NAME, TestLoggerExtension, project)
+        project.tasks.withType(Test).configureEach { task ->
```

Comment:
Would you mind to change this type of task configuration also in line 19? Otherwise you will end up creating all Test tasks directly during configuration time.

Reply:
Thank you. That was somehow missed out.